File: elixir/anagram/anagram.exs
defmodule Anagram do
  def match(base, candidates) do
    Enum.filter(candidates, fn(candidate) ->
      cond do
        String.downcase(base) === String.downcase(candidate) -> false
        format_string(base) === format_string(candidate)     -> true
        true -> false
      end
    end)
  end

  defp format_string(string) do
    String.downcase(string)
    |> String.graphemes
    |> Enum.sort
    |> Enum.join
  end
end

File: erlang/anagram/src/anagram.erl
-module(anagram).

-export([find/2, test_version/0]).

find(Word, Candidates) ->
  lists:filter(fun (C) -> is_anagram(Word, C) end, Candidates).

is_anagram(Word, Candidate) ->
  A = string:to_lower(Word),
  B = string:to_lower(Candidate),
  SortedA = lists:sort(A),
  SortedB = lists:sort(B),

  if A == B             -> false
   ; SortedA == SortedB -> true
   ; true               -> false
  end.

test_version() -> 1.
